libs/execution_context/ebpf_link.c

```diff
@@ -11,6 +11,7 @@ typedef struct _ebpf_link
 {
     ebpf_object_t object;
     ebpf_program_t* program;
+    ebpf_list_entry_t attach_list_entry;
```

Comment:
Move this to ebpf_object_t as and we can use the same path for map and link objects.

Currently, the program -> map is an array of pointers vs the program -> link is a linked list. We should treat them the same way.

Reply:
what list would that link belong to if its a field in the object_t type?

Reply:
It will still be in the list of links in ebpf_program struct.

Reply:
List head would still be in ebpf_program, but list entry would be in ebpf_object. This would give a common method for storing related objects.